File: nha-trang-bus-tour/src/main/java/com/grouppro/nhatrangbustour/Entity/Station.java
package com.grouppro.nhatrangbustour.Entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Nationalized;

import java.io.Serializable;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "Station")
@Table(name = "Station")
public class Station implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "StationID",
            updatable = false
    )
    private Long stationId;
    @Nationalized
    @Column(
            name = "StationName",
            nullable = false,
            length = 100
    )
    private String stationName;
    @Nationalized
    @Column(
            name = "StationLocation",
            nullable = false,
            length = 100
    )
    private String stationLocation;
    @OneToMany(fetch = FetchType.LAZY)
    @JsonBackReference
    private List<StationRoute> stationRoute;

}

File: swd392/lib/model/station.dart
class Station {
  int? stationId;
  String? stationName;
  String? stationLocation;

  Station({this.stationId, this.stationName, this.stationLocation});

  Station.fromJson(Map<String, dynamic> json) {
    stationId = json['stationId'];
    stationName = json['stationName'];
    stationLocation = json['stationLocation'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['stationId'] = this.stationId;
    data['stationName'] = this.stationName;
    data['stationLocation'] = this.stationLocation;
    return data;
  }
}